tests/versioncheck.rs

```diff
@@ -23,6 +23,12 @@ fn check_that_clippy_lints_has_the_same_version_as_clippy() {
 
 #[test]
 fn check_that_clippy_has_the_same_major_version_as_rustc() {
+    // do not run this test inside the upstream rustc repo:
+    // https://github.com/rust-lang/rust-clippy/issues/6683
+    if option_env!("RUSTC_TEST_SUITE").is_some() {
```

Comment:
Couldn't you just use ``cargo::is_rustc_test_suite()`` here? If not, just r=me.

Reply:
If I use `mod cargo;` to make the function available in `versioncheck.rs`, this would need `#![feature(once_cell)]` inside `versioncheck.rs` and causes a couple of `unused` warnings for `CARGO_TARGET_DIR` and `TARGET_LIB` which the test does not need.

Simply copying the function was the most straightforward way that I saw.